tools/clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -7962,6 +7962,8 @@ def warn_hlsl_barrier_no_mem_with_required_device_scope: Warning<
 def warn_hlsl_legacy_integer_literal_signedness: Warning<
   "literal value is treated as signed in HLSL 2021 and earlier, and unsigned in later language versions">,
   InGroup<HLSLLegacyLiterals>, DefaultIgnore;
+def err_hlsl_unsupported_semantic_index: Error<
+  "'%0' is defined with semantic index %1, but only values %2 through %3 are supported">;
```

Comment:
I assume the first value here is always `0` correct? If that's the case maybe it shouldn't be a variable. I realize that `%3` is currently only ever `7`, but we should absolutely add other verifications, but I think they would all be 0-?.

Reply:
Sounds right, done.